### ISSUE TITLE
Binding for app_metadata is missing

### ISSUE THREAD

**Issue by Yakimych** [MargaretKrutikova/bs-react-netlify-identity#1]
Since Netlify passes roles as an array in the `app_metadata` field on the `user` object, it would be great to be able to access it via reason bindings:
https://github.com/netlify/gotrue-js/blob/master/index.d.ts#L87

Here is the relevant Netlify documentation on roles:
https://www.netlify.com/docs/identity/#identity-user-metadata
```
"app_metadata": {"roles": ["investigator", "photographer"]}
```

**Comment by MargaretKrutikova** [comment]
I have added those bindings in `appMetaData` to the `user` object, could you try the branch `app-metadata` and  see whether it works for your setup?

**Comment by Yakimych** [comment]
Works great, thanks!

Just wondering - do `app_metadata` and `roles` have to be `options`? I guess it's the safest way, since the TS definitions say `any`?

**Comment by MargaretKrutikova** [comment]
Roles property must be optional, since it won’t exist on the metadata object if you don’t have any roles in netlify. 

I am not sure whether the whole “app_metadata” has to be optional though, is there anything in the netlify docs indicating that it should never be undefined on the JS side?

**Comment by Yakimych** [comment]
Oh, right - I was hoping roles would be an empty array. Not sure about `app_metadata` - the netlify docs seem to be a bit shallow on this. I guess the safest way is to have `option` for both.

**Comment by MargaretKrutikova** [comment]
Great, then we are good to go. Will publish a new version shortly 😄

**Comment by MargaretKrutikova** [comment]
Published in v1.0.4.
